### PR TITLE
 Sync between payment and opportunity allocations

### PR DESCRIPTION
**Pull request by jbrandolinisf**
# Critical Changes

# Changes
- We added some back-end syncing between Payments and Opportunity Allocations. This feature is disabled by default and should have no visible impact.

# Issues Closed

# New Metadata
Classes:
- ALLO_PaymentSync_TDTM
- ALLO_PaymentSync_SVC
- ALLO_PaymentSync_TEST

Custom Labels:
- alloPaidPmtsExceedsOpp
- alloRemoveOppWithPmt
- alloPaymentOverAllocatedError
# Deleted Metadata

### CODE REVIEW

**Comment by jbrandolinisf** [comment]
@ptewson-sfdo - Just tagging you to approve this, since we did the code review on the other branch.

**Comment by jbrandolinisf** [comment]
https://github.com/SalesforceFoundation/Cumulus/pull/4089

**Comment by jbrandolinisf** [comment]
@ceiroa - The Payment Allocations Two Way Sync PR...

**Comment by markross-sfdo** [comment]
@jbrandolinisf Is there a WI for this?

File: src/classes/ALLO_PaymentSync_SVC.cls [hunk 1]
```diff
@@ -253,7 +253,8 @@ public class ALLO_PaymentSync_SVC {
                 Allocation__c newOppAllocation = new Allocation__c(
                     General_Accounting_Unit__c = gau,
                     Opportunity__c = context.opportunity.id,
-                    Amount__c = newGAUAmtForOpp
+                    Amount__c = newGAUAmtForOpp,
+                    Percent__c = (context.Opportunity.Amount != 0) ? newGAUAmtFOrOpp / context.Opportunity.Amount * 100 : 0
```
Comment (pedrodalcol) [inline src/classes/ALLO_PaymentSync_SVC.cls]: This looks good! I would lowercase the 'o' in 'For'.

**Comment by melissabarber** [comment]
@jbrandolinisf are there any metadata changes we need to include in this PR for the release notes?

**Comment by jbrandolinisf** [comment]
@melissabarber - I added the new artifacts to the New Metadata section

**Comment by melissabarber** [comment]
Thanks, @jbrandolinisf!

**Comment by jbrandolinisf** [comment]
Superceded by #4290

**Pull request closed without merging**